access/handler.go

```diff
@@ -701,6 +713,192 @@ func (h *Handler) GetExecutionResultByID(ctx context.Context, req *access.GetExe
 	}, nil
 }
 
+// SubscribeBlocks handles subscription requests for blocks.
+// It takes a SubscribeBlocksRequest and an AccessAPI_SubscribeBlocksServer stream as input.
+// The handler manages the subscription to block updates and sends the subscribed block information
+// to the client via the provided stream.
+func (h *Handler) SubscribeBlocks(request *access.SubscribeBlocksRequest, stream access.AccessAPI_SubscribeBlocksServer) error {
+	// check if the maximum number of streams is reached
+	if h.StreamCount.Load() >= h.MaxStreams {
+		return status.Errorf(codes.ResourceExhausted, "maximum number of streams reached")
+	}
+	h.StreamCount.Add(1)
+	defer h.StreamCount.Add(-1)
+
+	startBlockID, startBlockHeight, blockStatus, err := h.getStartData(request.GetStartBlock(), request.GetBlockStatus())
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "invalid argument: %v", err)
+	}
+
+	sub := h.api.SubscribeBlocks(stream.Context(), startBlockID, startBlockHeight, blockStatus)
```

Comment:
What will happen if we get a failed subscription here?

Reply:
returns a new subscription that has already failed (NewFailedSubscription [here](https://github.com/UlyanaAndrukhiv/flow-go/blob/UlyanaAndrukhiv/subscribe-blocks/engine/access/subscription/subscription.go#L152)) or returns the error that caused the subscription to fail ([here](https://github.com/UlyanaAndrukhiv/flow-go/blob/UlyanaAndrukhiv/subscribe-blocks/engine/access/subscription/subscription.go#L54) ) which converts to rpc.ConvertError